internal/plugincommon/chain_support.go

```diff
@@ -61,6 +61,10 @@ func (c ccipChainSupport) DestChain() cciptypes.ChainSelector {
 }
 
 func (c ccipChainSupport) KnownSourceChainsSlice() ([]cciptypes.ChainSelector, error) {
+	if c.homeChain == nil {
+		return nil, fmt.Errorf("home chain is nil")
+	}
```

Comment:
Is this actually happening? Null pointer exceptions on `homeChain`? Feels like there should be a null check in the constructor in that case.